node_modules/col-assets/tests/test-bookmarklet.js

```diff
@@ -33,155 +34,97 @@ var UsersAPI = require('col-users');
-   */
-  it('is validated', function(callback) {
-    TestsUtil.getAssetLibraryClient(null, null, null, function(client, course, user) {
-      UsersTestUtil.assertGetMe(client, course, null, function(me) {
-        var bookmarkletClient = TestsUtil.getAnonymousClient();
-
-        // Too long title
-        AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, randomstring.generate(256), 'http://www.berkeley.edu/', null, 400, function() {
-
-          // Missing URL
-          AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', null, null, 400, function() {
-            // Invalid URL
-            AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'invalid url', null, 400, function() {
-              AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', '/invalidurl', null, 400, function() {
-                // Too long URL
-                AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/?q=' + randomstring.generate(229), null, 400, function() {
-
-                  // Invalid source
-                  AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/', {'source': 'invalid url'}, 400, function() {
-                    AssetsTestUtil.assertCreateLinkBookmarkletFails(bookmarkletClient, course, me.id, me.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/', {'source': '/invalidurl'}, 400, function() {
-
-                      return callback();
-                    });
-                  });
-                });
-              });
-            });
-          });
-        });
-      });
+  it('creates links with no optional metadata', function(callback) {
+    AssetsTestUtil.assertCreateLinkBookmarklet(bookmarkletClient, testCourse, testMe.id, testMe.bookmarklet_token, 'UC Davis', 'http://www.ucdavis.edu/', null, function(asset) {
+      callback();
+    });
+  });
+  it('defaults the link title to the URL', function(callback) {
+    var url = 'http://uci.edu';
+    AssetsTestUtil.assertCreateLinkBookmarklet(bookmarkletClient, testCourse, testMe.id, testMe.bookmarklet_token, null, url, null, function (asset) {
+      assert.equal(asset.title, url);
+      callback();
+    });
+  });
+  it('creates links with optional metadata', function(callback) {
+    var opts = {
+      'description': 'University of California, Berkeley homepage',
+      'source': 'http://www.universityofcalifornia.edu/uc-system'
+    };
+    AssetsTestUtil.assertCreateLinkBookmarklet(bookmarkletClient, testCourse, testMe.id, testMe.bookmarklet_token, 'UC Berkeley', 'http://www.berkeley.edu/', opts, function(asset) {
+      callback();
```

Comment:
@raydavis, I don't know why the `callback` naming convention was adopted as far as Mocha tests in particular. As a general rule, the convention in the codebase is that `callback` is used for passed-in callback functions, while `done` is used within functions if a different, non-conflicting callback needs to be defined. Usually this happens because of control flow with `async`, for example here: https://github.com/ets-berkeley-edu/suitec/blob/master/node_modules/col-activities/tests/util.js#L336